src/cloudant/design_document.py

```diff
@@ -27,7 +27,7 @@ class DesignDocument(Document):
     :class:`~cloudant.document.Document`.  A DesignDocument object is
     instantiated with a reference to a database and
```

Comment:
I think now that this work is splitting CouchDB and Cloudant design document functionality it probably makes sense to be more descriptive in this DesignDocument docstring and rework this docstring appropriately.  At the very least you probably should change the generic "database" reference to a link to the CouchDatabase class.